### PR TITLE
Add new feature which prints information about new plugins which can be upgraded to

### PR DESCRIPTION
**Pull request by jxr98**
**Make sure that you've checked the boxes below before you submit PR:**

### Always

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Written well with PR title, we generate the [release notes](https://github.com/jenkins-zh/jenkins-cli/releases) base on that

### For the bug fixes or features only

- [ ] [Quality Gate Passed](https://sonarcloud.io/dashboard?id=jenkins-zh_jenkins-cli). Change this URL to your PR.
- [ ] The coverage is xxx on the new lines
- [ ] I've tested it by manual in the following platform
  - [ ] MacOS
  - [x] Linux
  - [x] Windows
- [ ] Unit Test covered
- [ ] e2e Test covered

### CODE REVIEW

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Failed when I try to compile this PR:
```
codespace ➜ /workspaces/jenkins-cli (pr/wxharry/610) $ gh pr checkout 611
remote: Enumerating objects: 41, done.
remote: Counting objects: 100% (41/41), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 36 (delta 26), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (36/36), 8.79 KiB | 817.00 KiB/s, done.
From https://github.com/jenkins-zh/jenkins-cli
 * [new ref]         refs/pull/611/head -> plugincenter
Switched to branch 'plugincenter'
codespace ➜ /workspaces/jenkins-cli (plugincenter) $ make
go fmt ./util/...
go: github.com/creack/pty@v1.1.11: missing go.sum entry; to add it:
        go mod download github.com/creack/pty
make: *** [Makefile:108: fmt] Error 1
```

Still failed when I added the missing go.sum entry:
```
codespace ➜ /workspaces/jenkins-cli (plugincenter) $         go mod download github.com/creack/pty
codespace ➜ /workspaces/jenkins-cli (plugincenter ✗) $ make
go fmt ./util/...
go fmt ./client/...
go: downloading github.com/linuxsuren/http-downloader v0.0.22
go fmt ./app/...
go: downloading golang.org/x/text v0.3.5
go: downloading github.com/kevinburke/ssh_config v0.0.0-20190725054713-01f96b0aa0cd
go: downloading github.com/xanzy/ssh-agent v0.2.1
go: downloading github.com/sergi/go-diff v1.0.0
gofmt -s -w .
go vet ./...
go: downloading github.com/onsi/ginkgo v1.15.1
go: downloading github.com/onsi/gomega v1.11.0
app/cmd/plugin_center.go:9:2: missing go.sum entry for module providing package github.com/jedib0t/go-pretty/v6/table (imported by github.com/jenkins-zh/jenkins-cli/app/cmd); to add:
        go get github.com/jenkins-zh/jenkins-cli/app/cmd
/go/pkg/mod/github.com/google/go-github/v29@v29.0.3/github/git_commits.go:16:2: missing go.sum entry for module providing package golang.org/x/crypto/openpgp (imported by github.com/google/go-github/v29/github); to add:
        go get github.com/google/go-github/v29/github@v29.0.3
/go/pkg/mod/github.com/linuxsuren/go-cli-plugin@v0.0.4/pkg/cmd/fetch.go:7:2: missing go.sum entry for module providing package golang.org/x/crypto/ssh (imported by github.com/linuxsuren/go-cli-plugin/pkg/cmd); to add:
        go get github.com/linuxsuren/go-cli-plugin/pkg/cmd@v0.0.4
/go/pkg/mod/github.com/xanzy/ssh-agent@v0.2.1/sshagent.go:27:2: missing go.sum entry for module providing package golang.org/x/crypto/ssh/agent (imported by github.com/xanzy/ssh-agent); to add:
        go get github.com/xanzy/ssh-agent@v0.2.1
/go/pkg/mod/gopkg.in/src-d/go-git.v4@v4.13.1/plumbing/transport/ssh/auth_method.go:18:2: missing go.sum entry for module providing package golang.org/x/crypto/ssh/knownhosts (imported by gopkg.in/src-d/go-git.v4/plumbing/transport/ssh); to add:
        go get gopkg.in/src-d/go-git.v4/plumbing/transport/ssh@v4.13.1
/go/pkg/mod/github.com/jbenet/go-context@v0.0.0-20150711004518-d14ea06fba99/io/ctxio.go:16:2: missing go.sum entry for module providing package golang.org/x/net/context (imported by github.com/jenkins-zh/jenkins-cli/app/cmd); to add:
        go get github.com/jenkins-zh/jenkins-cli/app/cmd
/go/pkg/mod/gopkg.in/src-d/go-git.v4@v4.13.1/plumbing/transport/ssh/common.go:15:2: missing go.sum entry for module providing package golang.org/x/net/proxy (imported by gopkg.in/src-d/go-git.v4/plumbing/transport/ssh); to add:
        go get gopkg.in/src-d/go-git.v4/plumbing/transport/ssh@v4.13.1
/go/pkg/mod/github.com/mattn/go-isatty@v0.0.12/isatty_tcgets.go:6:8: missing go.sum entry for module providing package golang.org/x/sys/unix (imported by golang.org/x/term); to add:
        go get golang.org/x/term@v0.0.0-20210503060354-a79de5458b56
make: *** [Makefile:97: vet] Error 1
```

**Comment by jxr98** [comment]
All right. I am gonna fix this.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/jenkins-zh/jenkins-cli
 
-go 1.16
+go 1.14
```
Comment (LinuxSuRen) [inline go.mod]: why you change the go version here?

Comment (jxr98) [reply go.mod]: Oops, I've changed it back.

**Comment by LinuxSuRen** [comment]
hi @jxr98 Please pay attention to the conflicts.

**Comment by jxr98** [comment]
Hi @LinuxSuRen, I've solved the conflicts. By the way, any problem with this pr?

**Comment by jxr98** [comment]
To use this new feature, a plugin called 'pipeline restful api' needs to be installed in jenkins. I've tested using the command './jcli plugin center'.  It's a bit slow because the function is going to check every installed plugins in 'https://plugins.jenkins.io/api/plugin/'. The function only prints information about plugins which have newer versions then the installed.

**Comment by jxr98** [comment]
Hello Rick. This pr is the second needed to be reviewed.

**Pull request closed without merging**